### ISSUE TITLE
New Training Data for Gas Tank Level

### ISSUE THREAD

**Issue by nos86** [jomjol/AI-on-the-edge-device#1527]
### Files

Hello,
I'm trying to create my custom trained CNN for my gas tank.
Since the variation of the gauge is very slow and don't cover the full scale because the refill is done without emptying complete the tank; I decided to create pictures for training artificially.

In attachment the complete dataset of image (full size and already rescaled) for training. [data_all.zip](https://github.com/jomjol/AI-on-the-edge-device/files/10189405/data_all.zip)

I already used this set to train the network according to what reported in your other project (https://github.com/jomjol/neural-network-analog-needle-readout) but the results are quite bad with errors that exceed the 10% of final measurement.

Find below the expected deviation after training: 
![deviation](https://user-images.githubusercontent.com/6265510/206566494-59648c7c-41f6-454b-8e57-c63f2c3dbd4e.png)

Be aware that the number used for training don't much with gauge indication because the gauge is not linear while the number is uniformed distributed in a defined range (that is not 360° in order improve accuracy).

you can find more detail and training models (if you want) on my fork of your project: https://github.com/nos86/neural-network-analog-needle-readout

**Comment by jomjol** [comment]
Hi @nos86 ,
I really appreciate your effort. Looking quickly into your images, this is quite a taff job for the neural network. Very much details, black on black, no clear contrast in the resized image:
![grafik](https://user-images.githubusercontent.com/30766535/206756320-e1b71e85-9475-4996-97ef-a1d5d94b7324.png)
I assume, that there would be major changes needed in the cnn structure (size of input, internal structure, number of layers). I don't have the time and capacity to support you there personally.

To give you an estimation: I assume several hours of work for this single case if everything would go smooth (which never happens :-). A there are more requests with comparable task. That is nothing, which I can support in my free time.

If you have some time and knowledge, you can play around with this parameters. But be aware, that files, which are too large (> 1.5MB) will have problems on the ESP32.

I suggest to leave the issue open for a view days, maybe somebody else can support you.

**Comment by haverland** [comment]
You have only 100 images and most of them are from one image with a shadow of the original:
<img width="254" alt="image" src="https://user-images.githubusercontent.com/412645/206806265-0c0dbc57-106a-4c9b-84da-dca8a8ff4dff.png">

CNNs uses filters like border detection. So the shadow of the pointer of the original have a border more than the black pointer.

You should try to use real images to collect. And at least 10 for each output. If you not have enough for each of the 100 outcomes. Reduce the output to 10 at first step. 
You can raise up the images to 64x64. Look at Train_CNN_Analog-Readout_100-Small1_Dropout.ipynb. Here you can change the size in the first cell (input_shape = (64,64,3)
But it recognize with 128x128 not good. 
So collect, collect... Look on kaggle.com or others if you find datasets you can add.

At least: A usage of grayscale instead of rgb reduce not the size of the model in real (5-10%). Its because you reduce the first layer only.

**Comment by nos86** [comment]
Thank you so much for your help.

I'm not an expert of Computer Vision, but I'll read something and try to arrive to a conclusion.
Maybe, if the problem is the contrast, I can add some view steps before send image to CNN to improve the input and makes the CNN life easier.

**Comment by caco3** [comment]
Pleasereopen if needed
